### ISSUE TITLE
implicitAny doesn't work for 2.5.2

### ISSUE THREAD

**Issue by michael8090** [Microsoft/TypeScript#19416]
**TypeScript Version:**  2.5.2

**Code**

```ts
const a = [];
```

**Expected behavior:**
TS should show an error message when `noImplicitAny` is on;
**Actual behavior:**
No error message is shown, and `a` has a type of "any[]", as below:

![image](https://user-images.githubusercontent.com/2306105/31885105-01d47a40-b7b5-11e7-88d8-fa3f0e874e84.png)

**Comment by ahejlsberg** [comment]
This is working as intended. See #11432.

**Comment by michael8090** [comment]
@ahejlsberg You're right. Thanks
